src/browse.ts

```diff
@@ -312,12 +312,16 @@ export async function summarize (page: Page, keywords: string[], action: string)
     $('div').each(function () {
       if (keywords.length !== 0) {
         for (const keyword of keywords) {
-          if ($(this).children('div').length === 0 && $.html(this).toLowerCase().includes(keyword.toLowerCase())) {
+          if (hasNoNonTextChildren(this) && $.html(this).toLowerCase().includes(keyword.toLowerCase())) {
```

Comment:
This new function (`hasNoNonTextChildren`) makes sure that every child node of the element is just text and not an additional HTML element. This allows clicking on things like spans and divs that contain text in them, but nothing else.